threedi_modelchecker/checks/other.py

```diff
@@ -1082,3 +1068,77 @@ def _get_records(self, session):
                 )
             )
         )
+
+
+class UsedSettingsPresentCheck(BaseCheck):
+    def __init__(
+        self,
+        column,
+        settings_table,
+        filters=None,
+        level=CheckLevel.ERROR,
+        error_code=0,
+    ):
+        super().__init__(column, filters, level, error_code)
+        self.settings_table = settings_table
+
+    def get_invalid(self, session: Session) -> List[NamedTuple]:
+        # more than 1 row should be caught by another check
+        all_results = self.to_check(session).filter(self.column == True).all()
+        use_cols = len(all_results) > 0
+        if use_cols and session.query(self.settings_table).count() == 0:
+            return all_results
+        return []
+
+    def description(self) -> str:
+        return f"{self.column_name} in {self.table.name} is set to True but {self.settings_table.__tablename__} is empty"
+
+
+class SettingsLengthCheck(BaseCheck):
+    def __init__(
+        self,
+        column,
+        filters=None,
+        level=CheckLevel.ERROR,
+        error_code=0,
+        min_length=0,
+        max_length=1,
+    ):
+        super().__init__(column, filters, level, error_code)
+        self.min_length = min_length
+        self.max_length = max_length
+        self.observed_length = 0
+
+    def get_invalid(self, session: Session) -> List[NamedTuple]:
+        # return mock list in case the table is empty when it shouldn't be
+        all_results = self.to_check(session).all()
+        self.observed_length = len(all_results)
+        if (self.observed_length > self.max_length) or (
+            self.observed_length < self.min_length
+        ):
+            return all_results if self.observed_length > 0 else ["foo"]
+        else:
+            return []
+
+    def description(self) -> str:
+        return (
+            f"{self.table.name} has {self.observed_length} rows, "
+            f"but should have at least {self.min_length} rows and "
+            f"at most {self.max_length} rows."
+        )
+
+
+class AggregationSettingsInvervalCheck(BaseCheck):
+    def get_invalid(self, session: Session) -> List[NamedTuple]:
+        return (
+            session.query(models.AggregationSettings)
+            .join(
+                models.TimeStepSettings,
+                models.AggregationSettings.interval
```

Comment:
The second argument of `.join` should be the "ON" clause. So this will read in SQL : `SELECT * FROM aggregation_settings JOIN timestep_settings ON aggregation_settings.interval < timestep_settings.output_time_step`. The end result in a weird way OK but you should use a where clause. Also the join itself is special because there is nothing to join on, but both tables have (at most) 1 record. I think in general you must be explicit that it is a cartesian join. SQLAlchemy will emit warnings if you don't (see https://docs.sqlalchemy.org/en/20/changelog/migration_14.html#built-in-from-linting-will-warn-for-any-potential-cartesian-products-in-a-select-statement)